sparkmagic/proposal.md

```diff
@@ -0,0 +1,49 @@
+# Proposal for Incubation
+
+## Subproject name
+
+sparkmagic
+
+## Development team and Advocate
+
+Subproject development team:
+
+* Alejandro Guerrero Gonzalez, Microsoft (`@aggFTW`)
+* Joel Zambrano, Microsoft (`@djoelz`)
+* Auberon Lopez, Cal Poly (`@alope107`)
+
+Steering Council Advocate
+
+* Brian Granger (`@ellisonbg`)
+
+## Subproject goals, scope and functionality
+
+Goals:
+* Make Spark usage from the IPython notebook as frictionless as possible by decoupling Spark code execution from Spark installation.
+* These magics will be packaged so that they can be pip-installable or conda-installable.
+* Allow any Jupyter installation to point to different Spark installations.
+
+Scope:
+* Spark will be the first back end provided by these magics, but it could easily be extended to other big data back ends.
+* Remote Spark execution through a REST endpoint which allows for Scala, Python, and R support as of Sept 2015.
+
+Functionality:
+* Enable users to remotely connect to a REST Spark back end.
+* Return of pandas dataframes from Spark when appropriate. This will allow users to integrate spark with the Python libraries available for pandas dataframes.
+
+## Audience
+
+Spark users who want to have the best experience possible from a Notebook. These users care about the libraries ecosystem the Jupyter community has created.
+
+## Other options
+
+Alternatives we know of:
+
+* Combination of IPython, R kernel (or rpy2), and Scala kernel for an in-Spark-cluster Jupyter installation. This does not allow the user to point to different Spark clusters. It might also result in resource contention (CPU or memory) between the Jupyter installation and Spark.
+* IBM's Spark kernel does not provide a REST endpoint, requires the installation of Jupyter in the cluster, and does not create pandas dataframes.
```

Comment:
Yup, great clarification, totally with you. Was mainly commenting re: the resource allocation issue. Totally with you on this being a pain point, that's especially been true for our EC2 deployments, maybe less so for deployments in a private, managed HPC environment.

One thing I'm curious about for the RESTful model would be how large aggregations behave, i.e. when a large result needs to be returned to the driver. Do you have experience with this? Maybe that's what you already meant by 

> Compute driver (kernel) stays close to compute workers (Spark) stays close to (big) data